azkaban-common/src/main/java/azkaban/utils/AzkabanPatternLayout.java

```diff
@@ -0,0 +1,112 @@
+package azkaban.utils;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import org.apache.log4j.helpers.PatternConverter;
+import org.apache.log4j.helpers.PatternParser;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+
+
+/**
+ * When we use the log4j Kafka appender, it seems that the appender simply does not log the stack trace anywhere
+ * Seeing as the stack trace is a very important piece of information, we create our own PatternLayout class that
+ * appends the stack trace to the log message that reported it, so that all the information regarding that error
+ * can be found one in place. In addition, we create a new PatternParser to deliver logs with current host name.
+ */
+public class AzkabanPatternLayout extends PatternLayout {
+  protected String host;
+
+  public AzkabanPatternLayout(String s) {
+    super(s);
+  }
+
+  public AzkabanPatternLayout() {
+    super();
+  }
+
+  // TODO: There is no easy way to get Jetty Server's port by InetAddress API.
+  // We might need to find some other way to find this application's port
+  protected String getPort() {
+    return "";
+  }
+
+  protected String getHostname() {
+    if (host == null) {
+      try {
+        InetAddress addr = InetAddress.getLocalHost();
+        this.host = addr.getHostName();
+      } catch (UnknownHostException e) {
+
+        // Prevent exposing exceptions when something goes wrong.
+        this.host = "localhost";
+      }
+    }
+    return host;
+  }
+
+  @Override
+  protected PatternParser createPatternParser(String pattern) {
+    return new PatternParser(pattern) {
+
+      @Override
+      protected void finalizeConverter(char c) {
+        PatternConverter pc = null;
+
+        switch (c) {
```

Comment:
I intended to create the switch structure to help more options being added easily in future. Will might have AZ version, AZ port attached to every log message.